lang/funcs/filesystem.go

```diff
@@ -212,25 +213,33 @@ func MakeFileExistsFunc(baseDir string) function.Function {
 func MakeFileSetFunc(baseDir string) function.Function {
 	return function.New(&function.Spec{
 		Params: []function.Parameter{
+			{
+				Name: "path",
+				Type: cty.String,
+			},
 			{
 				Name: "pattern",
 				Type: cty.String,
 			},
 		},
 		Type: function.StaticReturnType(cty.Set(cty.String)),
 		Impl: func(args []cty.Value, retType cty.Type) (cty.Value, error) {
-			pattern := args[0].AsString()
-			pattern, err := homedir.Expand(pattern)
+			path := args[0].AsString()
+			pattern := args[1].AsString()
+
+			path, err := homedir.Expand(path)
```

Comment:
Sorry I didn't catch this on the first look, but... I see that we've been a bit inconsistent about whether a given path is subject to "homedir" expansion, with `fileexists` doing it but the `file` and `templatefile` functions _not_ doing it.

I'd lean towards _not_ doing it here for consistency with `file` and `templatefile`, because otherwise it's a bit of a hidden microsyntax that is likely to be surprising. If we find that `fileset` against files in the home directory is a common situation (which seems unlikely to me right now) then I'd prefer to meet that use case by adding an explicit `path.home` so that it would be consistent with how we handle all of the other types of base directory, like `"${path.home}/foo/bar"`. But the expected way to use these functions is against `path.module`, so I think we should wait until we find compelling use-case for it before adding it.

Reply:
Removed in 9eaf7eb9cbab64ec745cd5efbd3980cab10d2ac8 👍